python/graphstorm/gconstruct/transform.py

```diff
@@ -1609,8 +1609,12 @@ def process_features(data, ops: List[FeatTransform], ext_mem_path=None):
                         if key in new_data else val
                     new_data[key] = val
 
-        # Write feature dimension back to the feature config
-        feat_dim_dict[op.feat_name] = op.feat_dim
+            # Write feature dimension back to the feature config
+            if op.feat_name not in feat_dim_dict:
+                feat_dim_dict[op.feat_name] = op.feat_dim
+            else:
+                feat_dim_dict[op.feat_name] = (op.feat_dim[0] + feat_dim_dict[op.feat_name][0],) + \
+                                        feat_dim_dict[op.feat_name][1:]
```

Comment:
Can we clarify what's happening here and the expected input/output? These are supposed to be integer values right?

Reply:
For two transformed value with Dimension (M, K, ..) and (N, K, ..), we directly concatenate the value, so the output will be (M + N, K, ..)

Reply:
Added comment for more explanation.